rmf_fleet_adapter_python/scripts/rmf_fleet_adapter_python/test_adapter.py

```diff
@@ -0,0 +1,594 @@
+    at_least_one_incomplete = False
+    completed = False
+
+    def task_cb(msg):
+        nonlocal at_least_one_incomplete
+        nonlocal completed
+
+        if msg.state == TASK_STATE_COMPLETED:
+            completed = True
+            at_least_one_incomplete = False
+        else:
+            completed = False
+            at_least_one_incomplete = True
+
+    task_node = rclpy.create_node("task_summary_node")
+    task_node.create_subscription(TaskSummary,
+                                  "task_summaries",
+                                  task_cb,
+                                  10)
+
+    # INIT DISPENSERS =========================================================
+    quiet_dispenser = MockQuietDispenser(quiet_dispenser_name)
+    flaky_dispenser = MockFlakyDispenser(flaky_dispenser_name)
+
+    # FINAL PREP ==============================================================
+    rclpy_executor = SingleThreadedExecutor()
+    rclpy_executor.add_node(task_node)
+    rclpy_executor.add_node(cmd_node)
+    rclpy_executor.add_node(quiet_dispenser)
+    rclpy_executor.add_node(flaky_dispenser)
+
+    last_quiet_state = False
+    last_flaky_state = False
+
+    # GO! =====================================================================
+    adapter.start()
+
+    print("# SENDING NEW REQUEST ############################################")
+    request = adpt.type.CPPDeliveryMsg("test_delivery",
+                                       pickup_name,
+                                       quiet_dispenser_name,
+                                       dropoff_name,
+                                       flaky_dispenser_name)
+    quiet_dispenser.reset()
+    flaky_dispenser.reset()
+    adapter.request_delivery(request)
+    rclpy_executor.spin_once(1)
+
+    for i in range(1000):
+        if quiet_dispenser.success_flag != last_quiet_state:
+            last_quiet_state = quiet_dispenser.success_flag
+            print("== QUIET DISPENSER FLIPPED SUCCESS STATE ==",
+                  last_quiet_state)
+
+        if flaky_dispenser.success_flag != last_flaky_state:
+            last_flaky_state = flaky_dispenser.success_flag
+            print("== FLAKY DISPENSER FLIPPED SUCCESS STATE ==",
+                  last_flaky_state)
+
+        if quiet_dispenser.success_flag and flaky_dispenser.success_flag:
+            rclpy_executor.spin_once(1)
+            break
+
+        rclpy_executor.spin_once(1)
+        time.sleep(0.2)
+
+    print("\n== DEBUG TASK REPORT ==")
+    print("Visited waypoints:", robot_cmd.visited_waypoints)
+    print("At least one incomplete:", at_least_one_incomplete)
+    print("Completed:", completed)
+    print()
+
+    assert len(robot_cmd.visited_waypoints) == 6
+    assert all([x in robot_cmd.visited_waypoints for x in [0, 5, 6, 7, 8, 10]])
+    assert at_least_one_incomplete
+
+    # Uncomment this to send a second request.
+    # But note! The TaskManager has to be fixed first to allow task pre-emption
```

Comment:
Seems like this code is not ready. Add a TODO (or whatever the standard for todos is) so we can search for it